app/src/main/java/com/android/bookswap/ui/map/Map.kt

```diff
@@ -96,116 +100,121 @@ fun MapScreen(
     geolocation.startLocationUpdates()
     cameraPositionState.position =
         CameraPosition.fromLatLngZoom(LatLng(latitude.value, longitude.value), INIT_ZOOM)
+    isOnline = isNetworkAvailable(context)
   }
-  // Stop location and books updates when the screen is disposed
-  DisposableEffect(Unit) {
-    onDispose {
-      geolocation.stopLocationUpdates()
-      bookManagerViewModel.stopUpdatingBooks()
+  if (!isOnline) {
+    Text("No internet connection")
+  } else {
```

Comment:
This condition will compose a blank screen with a simple Text when offline, without the navigation and top app bar, and prevents the user from using the app in offline mode entirely.
The only useage of internet on this screen is when calling `bookManagerViewModel.startUpdatingBooks()`. Thus when offline, we can simply skip this call and display a Toast message such as "No internet connexion" instead.